modules/internal/ChapelRange.chpl

```diff
@@ -792,7 +851,7 @@ module ChapelRange {
      partially bounded, the return value will not (cannot) be a
      `param`.
   */
-  proc range.hasLast() param where !stridable && !hasLowBound()
+  proc range.hasLast() param where !stridable && !(hasLowBound() && hasHighBound())
     return hasHighBound();
```

Comment:
I think this overload's where clause should be simply `where !stridable`.

Reply:
I'd hoped this during development as well, but don't believe it's the case (at least not without other additional changes).  For example, making this change causes `(4..3).hasFirst()` or `.hasLast()` to return `true` where the correct answer is `false`.